cloudevents/tests/test_http_cloudevent.py

```diff
@@ -57,6 +81,16 @@ def test_http_cloudevent_equality(dummy_attributes, my_dummy_data, your_dummy_da
     assert event1 != event2 and event3 != event1
 
 
+def test_http_cloudevent_equality_must_not_throw(dummy_event, non_cloudevent_value):
+    assert isinstance(dummy_event == non_cloudevent_value, bool)
+
+
+def test_http_cloudevent_must_not_equal_to_non_cloudevent_value(
+    dummy_event, non_cloudevent_value
+):
+    assert not dummy_event == non_cloudevent_value
+
```

Comment:
You're utilizing a non-obvious nature of the `params` config here. I assume you'd like to have different parameters being used in order to ensure that equality works as needed for different types, right?

In this case, you should use parameterized tests over customizing the fixture behavior.

Also, the `not_equal` test case ensures as well that no errors are thrown. I don't see a benefit of having a second test here.

I'd go with the following instead of using `params` for the `fixture`:

```python
@pytest.mark.parametrize("value_to_check,expected", [(1, False), (None, False), (object(), False), ("Hello World", False"])
def test_http_cloudevent_equality(dummy_event: CloudEvent, value_to_check: typings.Any, expected_result: bool) -> None:
    assert (dummy_event == value_to_check) == expected
```

P.S. the code needs to be pretty-formatted as I've just written it in GH editor and haven't tested.